subprojects/libft/libft.h

```diff
@@ -144,4 +124,69 @@ char			**ft_strfoversplit(const char *s, int should_split(int c));
 bool			ft_getopt(struct s_ft_getopt *opt, int argc, char **argv,
 					const char *optstring);
 
+/*
+** LINKED LIST
+*/
+
+/*
+** list - double linked list implementation.
+**
+** This linked list implementation has been inspired by the list.h from the
+** Linux kernel.
+**
+** The s_list2_conn struct contains pointers to the next and previous
+** s_list2_conn structs. The functions contained in this header will malipulate
+** these pointers. The struct contains only those pointers and therefor no
+** content. It should not be used standalone. Instead, put it inside of a
+** parrent struct as follows:
+**
+** struct s_example_node
+** {
+**   char         *foo;
+**   int          bar;
+**   [etc...]
+**   t_list2_conn conn;
+** }
+**
+** To insert an element, malloc the aforementioned struct and give list2_insert
+** a pointer to the conn member of the struct.
+**
+** To get the s_example_node struct from a conn, an unpack function must be
+** written. In this case the function would look like this:
+**
+** struct s_state_option    *unpack_example(struct s_list2_conn *conn)
+** {
+**     if (conn == NULL)
+**         return (NULL);
+**     return (struct s_example_node *)(
+**         (char *)conn - offsetof(struct s_example_node, conn));
+** }
+**
+** This function takes a pointer to the conn member of the example struct, and
+** returns a pointer to the example struct itself. It does this by calculating
+** the offset of the conn member relative to the start of the example struct.
+*/
+
+typedef struct	s_list_conn
+{
+	struct s_list_conn	*prev;
+	struct s_list_conn	*next;
+}				t_list_conn;
+
+typedef struct	s_list_meta
+{
+	struct s_list_conn	*first;
+	struct s_list_conn	*last;
+	size_t				len;
+}				t_list_meta;
+
+void			ft_list_insert(
+					t_list_meta *meta,
+					t_list_conn *prev,
+					t_list_conn *new);
+
+void			ft_list_unlink(
+					t_list_meta *meta,
+					t_list_conn *conn);
```

Comment:
Is not listing the first argument on the same line as the function name the current fashion?

Reply:
Personally I like them either all on the same line or all of them separate on the next lines indented one extra relative to the function name. Having only one argument next to the function name looks jarring because the vertical position of the parameters changes.

Although I'm not sure we can get anywhere discussing minor style preferences. It's just that - a preference.

That said, it looks more ugly on the GitHub UI because it renders tabs with a width of 8 units. I should add an https://editorconfig.org/ to the repo so github renders it correctly.